File: ppbluetoothkit/src/main/java/com/peng/ppscale/data/PPBodyDetailInfoModel.kt
package com.peng.ppscale.data

data class PPBodyDetailInfoModel(
    //唯一键 ->"bodyParamKey": "ppBMI",
    var bodyParamKey: String = "",
    // 名称
    var bodyParamNameString: String = "",
    // 对名称的详细说明
    var introductionString: String = "",
    // 范围节点 [0,18.5,25,30,42]
    var standardArray: List<Float>? = ArrayList(),
    // 当前值 没有单位
    var currentValue: Float = 0.0f,
    //ui展示的值,经过计算的或者四色五人后的值
    var currentValueStr: String = "",
    // 当前值对应的单位
    var unit: String = "",
    // 当前值坐落的区间
    var currentStandard: Int = 0,
    // 对节点区间的描述 (瘦,普通,偏胖,肥胖)
    var standardTitleArray: List<String> = ArrayList(),
    //标准的名称
    var standardTitle: String = "",
    // 节点区间的颜色 (#F5A623,#14CCAD,#F43F31,#C23227)
    var colorArray: List<String> = ArrayList(),
    var standColor: String = "",
    // 对当前所在区间的评价(BMI_leve1_evaluation,BMI_leve2_evaluation,BMI_leve3_evaluation,BMI_leve4_evaluation)
    var evaluationArray: List<String> = ArrayList(),
    //当前的评价
    var standeEvaluation: String = "",
    // 对当前所在区间的建议(BMI_leve1_suggestion,BMI_leve2_suggestion,BMI_leve3_suggestion,BMI_leve4_suggestion)
    var suggestionArray: List<String> = ArrayList(),
    //需注意的区间 [1,2]
    var watchfulArray: ArrayList<Int> = ArrayList(),
    //当前的建议
    var standSuggestion: String = "",
    // 是否有区间
    var hasStandard: Boolean = false
):java.io.Serializable

File: ppbluetoothkit/src/main/java/com/peng/ppscale/ios/PPBodyDetailInfoModel.m
//
//  PPBodyDetailInfoModel.m
//  PPBluetoothKit
//
//  Created by 彭思远 on 2023/6/15.
//

#import "PPBodyDetailInfoModel.h"
#import "PPAstonExtension.h"
#import "PPBodyDetailStandardArray.h"


@interface PPBodyDetailInfoModel()



// 对当前所在区间的评价(BMI_leve1_evaluation,BMI_leve2_evaluation,BMI_leve3_evaluation,BMI_leve4_evaluation)
@property (nonatomic,strong) NSArray <NSString *> *evaluationArray;

// 对当前所在区间的建议(BMI_leve1_suggestion,BMI_leve2_suggestion,BMI_leve3_suggestion,BMI_leve4_suggestion)
@property (nonatomic,strong) NSArray <NSString *> *suggestionArray;

@end

@implementation PPBodyDetailInfoModel



- (instancetype)initWithDic:(NSDictionary *)dic
                 andKeyName:(NSString *)key
              standardArray:(NSArray <NSNumber *>*)standardArray
               currentValue:(CGFloat)currentValue
{
    
    self = [super init];
    
    if (self){
        
        NSDictionary *d = [dic valueForKey:key];
        self.bodyParamKey = [d valueForKey:@"bodyParamKey"];;
        self.suggestionArray = [d valueForKey:@"suggestionArray"];
        self.evaluationArray = [d valueForKey:@"evaluationArray"];
        self.standardTitleArray = [d valueForKey:@"standardTitleArray"];
        self.introductionString = [d valueForKey:@"introductionString"];
        self.bodyParamNameString = [d valueForKey:@"bodyParamNameString"];
        self.colorArray = [d valueForKey:@"colorArray"];
        self.watchfulArray = [d valueForKey:@"watchfulArray"];
        self.currentUnit = [d valueForKey:@"currentUnit"];
        
        //当前值
        self.currentValue = currentValue;
        // 是否有区间标准
        BOOL hasStandard = standardArray.count > 0 ? YES:NO;
        
        if (hasStandard){
            
            [self fillWithStandardArray:standardArray currentValue:currentValue];

        }else{
            self.watchfulArray = @[];
            self.standardArray = @[];
            self.currentStandard = 0;
            self.standardTitle = @"";
            self.standColor = @"";
            self.standSuggestion = @"";
            self.standeValuation = @"";
        }
        
    }
    
    return self;
    
}

- (void)fillWithStandardArray:(NSArray <NSNumber *>*)standardArray
                        currentValue:(CGFloat)currentValue{
 
    //范围节点
    self.standardArray = standardArray;
   
    // 当前值坐落的区间
    NSInteger currentStandard = [PPBodyDetailStandardArray calculateCurrentStandardWithValue:currentValue andStandardArray:standardArray];
    self.currentStandard = currentStandard;
    // 对节点区间的描述
    self.standardTitle = [self safeFetchListValueByIndex:currentStandard andList:self.standardTitleArray];
    // 节点区间的颜色
    self.standColor = [self safeFetchListValueByIndex:currentStandard andList:self.colorArray];
    // 对当前所在区间的评价
    self.standSuggestion = [self safeFetchListValueByIndex:currentStandard andList:self.suggestionArray];
    // 对当前所在区间的建议
    self.standeValuation = [self safeFetchListValueByIndex:currentStandard andList:self.evaluationArray];
    
    
}

- (NSString *)safeFetchListValueByIndex:(NSInteger)index andList:(NSArray <NSString *>*)list{
    
    if (index < list.count){
        return [list objectAtIndex:index];
    }

    return list.firstObject;
}

- (NSString *)description{
    
    return [self PPAston_JSONString];
}

@end
